### PR TITLE
initial commit

### PR DESCRIPTION
**Pull request by Ohen**
initial review

with [island.js](https://github.com/spearhead-ea/island/tree/feat.use.status.exporter)

RELEASE_CANDIDATE

### CODE REVIEW

**Comment by belldoor** [comment]
You should tag `RELEASE_CANDIDATE` on commit message instead of Pull Request message.

File: src/status-exporter.ts [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+}
+
+function makeFileName(args: { name?: string, hostname?: string, servicename?: string }): string {
+  if (args.name && !args.name.match('status.json')) {
+    return args.name + '.status.json';
+  }
+  if (!args.name && (args.hostname || args.servicename)) {
+    args.name = _.values(_.pickBy(args, (value, key) => {
+      return Boolean(value);
+    })).join('.') + '.status.json';
```
Comment (Ohen) [inline src/status-exporter.ts]:
```ts
[args.hostname, args.servicename].filter(Boolean).join('.')
```

Comment (Ohen) [inline src/status-exporter.ts]: [reflect it](https://github.com/spearhead-ea/island-status-exporter/pull/1/files#diff-9074ba8434db9fd7a8e8bfdc781bb7d3R60)

File: src/status-exporter.ts [hunk 2]
```diff
@@ -0,0 +1,176 @@
...
+  }
+  return PATH + '/' + (args.name || 'status.json');
+}
+
+async function lookupProcess(): Promise<any> {
+  return new Promise((res, rej) => {
+    pidusage.stat(process.pid, (err, result) => {
+      if (err) {
+        throw new Error('Failed lookup process information' + err);
```
Comment (Ohen) [inline src/status-exporter.ts]: [reflect it](https://github.com/spearhead-ea/island-status-exporter/pull/1/files#diff-9074ba8434db9fd7a8e8bfdc781bb7d3R69)

File: src/status-exporter.ts [hunk 3]
```diff
@@ -0,0 +1,176 @@
...
+      calculatedData.memory = setDecimalPoint((result.memory / 1024 / 1024)); // MB
+      return res();
+    });
+  });
+}
+
+export namespace StatusExporter {
+  export function initialize(using: boolean, option: { name?: string, hostname?: string, servicename?: string }) {
+    HOSTNAME = option.hostname;
+    STATUSEXPORT = using;
```
Comment (Ohen) [inline src/status-exporter.ts]: [reflect It](https://github.com/spearhead-ea/island-status-exporter/pull/1/files#diff-9074ba8434db9fd7a8e8bfdc781bb7d3R80)

File: src/status-exporter.ts [hunk 4]
```diff
@@ -0,0 +1,176 @@
...
+      collectedData[type].requestCount++;
+      if (time) collectedData[type].receviedTime = time;
+      return;
+    }
+    if (!collectedData[type].requestCount) {
+      collectedData[type].requestCount = 1;
+      collectedData[type].requestStartedAt = +new Date();
+      if (time) collectedData[type].receviedTime = time;
+      return;
+    }
```
Comment (Ohen) [inline src/status-exporter.ts]: [reflect It](https://github.com/spearhead-ea/island-status-exporter/pull/1/files#diff-9074ba8434db9fd7a8e8bfdc781bb7d3R143)

File: src/status-exporter.ts [hunk 5]
```diff
@@ -0,0 +1,176 @@
...
+
+    collectedData[type].requestCount++;
+    if (time && collectedData[type].receviedTime) collectedData[type].receviedTime += time;
```
Comment (Ohen) [inline src/status-exporter.ts]: [변수명 명확 하게 알 수 있도록 변경](https://github.com/spearhead-ea/island-status-exporter/pull/1/files#diff-9074ba8434db9fd7a8e8bfdc781bb7d3R23)

File: src/status-exporter.ts [hunk 6]
```diff
@@ -0,0 +1,164 @@
...
+    await fs.writeFileSync(SAVE_FILE_NAME + '.tmp', JSON.stringify(calculatedData, null, 2), 'utf8');
+    return await moveJsonFile(SAVE_FILE_NAME + '.tmp', SAVE_FILE_NAME);
+  }
+
+  async function calculateMeasurements() {
+    if (!STATUSEXPORT) return;
+    const tmpData: { [type: string]: CollectData } = _.clone(collectedData);
+    clearData();
+    await lookupProcess();
+    return await _.forEach(tmpData, async (value, type) => {
```
Comment (heycalmdown) [inline src/status-exporter.ts]: this `await` do nothing

File: src/status-exporter.ts [hunk 7]
```diff
@@ -0,0 +1,164 @@
...
+      collectedData[type].executedCount++;
+      collectedData[type].totalExecutionTime = time;
+      return;
+    }
+    if (!collectedData[type].executedCount) {
+      collectedData[type].executedCount = 1;
+      collectedData[type].executionStartedAt = +new Date();
+      collectedData[type].totalExecutionTime = time;
+      return;
+    }
```
Comment (heycalmdown) [inline src/status-exporter.ts]: this block seems redundant

File: src/status-exporter.ts (hunk 6 above)
Comment (Ohen) [inline src/status-exporter.ts]: remove it https://github.com/spearhead-ea/island-status-exporter/pull/1/commits/5460a735234d6efeb8ca3d63818ac830c4907785

File: src/status-exporter.ts (hunk 7 above)
Comment (Ohen) [inline src/status-exporter.ts]: modify redundant block https://github.com/spearhead-ea/island-status-exporter/pull/1/commits/5460a735234d6efeb8ca3d63818ac830c4907785

File: package.json [hunk 8]
```diff
@@ -0,0 +1,39 @@
...
+  "main": "dist/status-exporter.js",
+  "typings": "dist/status-exporter.d.ts",
+  "author": "ohen <matoda88@gmail.com>",
+  "contributors": [],
+  "scripts": {
+    "prepublish": "gulp build",
+    "build": "gulp build",
+    "watch": "gulp build && gulp watch",
+    "clean": "gulp clean",
+    "test": "gulp build && DEBUG=* jasmine",
```
Comment (UniverseHan) [inline package.json]: 여기에도 저대신 하핫
ISLAND_LOGGER_LEVEL=crit

Comment (Ohen) [inline package.json]: 반영해두었습니다

Comment (UniverseHan) [inline package.json]: 감사합니다!!

**Pull request merged**